src/inventory/transaction/action/SlotChangeAction.php

```diff
@@ -74,6 +75,13 @@ public function validate(Player $source) : void{
 		if($this->targetItem->getCount() > $this->inventory->getMaxStackSize()){
 			throw new TransactionValidationException("Target item exceeds inventory max stack size");
 		}
+		if($this->inventory instanceof SlotSafeInventory && !$this->targetItem->isNull()){
+			foreach($this->inventory->getSlotValidators() as $validator){
+				if(!$validator($this->inventory, $this->targetItem, $this->inventorySlot)){
```

Comment:
Perhaps the validators should throw (or return) TransactionValidationException directly, so that a detailed error can be given

Reply:
I think throwing an exception is better? If it returns false, then it's a general error.

Reply:
We can't document a Closure to throw a specific type of exception. However, we could have it return `TransactionValidationException|null`.

Reply:
So people is force to return an exception with an error message

Reply:
Yes